alfred/files/alfred.init

```diff
@@ -103,41 +94,36 @@ alfred_start()
 	return 0
 }
 
-start()
-{
+start_service() {
 	config_load "alfred"
 	config_foreach alfred_start alfred
-
-	if [ "$enable" = "0" ]; then
-		exit 0
-	fi
-
-	echo "${initscript}: starting alfred"
-	SERVICE_PID_FILE="$pid_file_alfred"
-	service_start /usr/sbin/alfred ${alfred_args}
-
-	if [ "$vis_enable" = "1" ]; then
-		echo "${initscript}: starting batadv-vis"
-		SERVICE_PID_FILE="$pid_file_vis"
-		service_start /usr/sbin/batadv-vis ${vis_args}
-	fi
-
-	if [ "$run_facters" = "1" ]; then
+	procd_open_instance "alfred"
+	procd_set_param command /usr/sbin/alfred
+	procd_append_param command ${alfred_args}
+	procd_close_instance
+
+	[ "$vis_enable" == "1" ] && {
+		procd_open_instance "batadv-vis"
+		procd_set_param command /usr/sbin/batadv-vis
+		procd_append_param command ${vis_args}
+		procd_close_instance
+	}
+
+	[ "$run_facters" == "1" ] && {
 		( for file in $facters_dir/* ; do [ -x $file ] && $file ; done )
 		if ! ( grep -q "for file in $facters_dir/\* ; do " /etc/crontabs/root 2>/dev/null ) ; then
 			echo "*/5 * * * * ( for file in $facters_dir/* ; do [ -x \$file ] && \$file ; done )" >> /etc/crontabs/root
 			/etc/init.d/cron enable
 			/etc/init.d/cron restart
 		fi
-	fi
+	}
+}
+
+service_triggers() {
+    procd_add_reload_trigger "alfred"
 }
 
-stop()
-{
-	SERVICE_PID_FILE="$pid_file_alfred"
-	service_stop /usr/sbin/alfred
-	SERVICE_PID_FILE="$pid_file_vis"
-	[ -x /usr/sbin/batadv-vis ] && service_stop /usr/sbin/batadv-vis
+stop_service() {
 	sed "\|for file in $facters_dir/\* ; do |d" -i /etc/crontabs/root
```

Comment:
I get an error here because /etc/crontabs/root does not exist. Maybe add 2> /dev/null or check if the file exists?